### DISCUSSION TITLE
How to make library only render at client

### DISCUSSION THREAD

**Discussion by tienlx97** [vercel/next.js#55644] [App Router]
I use symlink `internal` facebook.com "react-dom" to nextjs. It works almost well (90% ). But the internal react-dom have another function name : `unstable_createEventHandle` . `unstable_createEventHandle` will handle some event: fosusin, focusout, hover, ... And this function need run on client . So how to make it run only on client

```tsx
/* function Description:
 We cannot support arbitrary native events with eager root listeners  because the eager strategy relies on knowing the whole list ahead of time. If we wanted to support this, we'd have to add code to keep track  (or search) for all portal and root containers, and lazily add listeners to them whenever we see a previously unknown event. This seems like a lot  of complexity for something we don't even have a particular use case for. Unfortunately, the downside of this invariant is that *removing* a native event from the list of known events has now become a breaking change for any code relying on the createEventHandle API.
*/
 unstable_createEventHandle("focusin", {
     passive : undefined
 })
```

This is the react-dom that I fb use
[Link](https://github.com/facebook/react/blob/builds/facebook-www/compiled/facebook-www/ReactDOM-dev.modern.js#L46195)

**Comment by icyJoseph** [comment]
- https://demystifying-rsc.vercel.app/client-components/no-ssr/

**Comment by VladSez** [comment]
You probably have to use dynamic-import
https://nextjs.org/docs/app/building-your-application/optimizing/lazy-loading#skipping-ssr

https://x.com/asidorenko_/status/1701574945795408009?s=46&t=_RN2fQnPTv5buAq00Oxwaw

**Reply by tienlx97** [reply]
I think the problem is next.js use default react-dom package. So when I log `react-dom`. I does't have my function `unstable_createEventHandle`. [This is default nextjs react-dom-experimental](https://github.com/vercel/next.js/tree/canary/packages/next/src/compiled/react-dom-experimental)
